collab/src/lock/mod.rs

```diff
@@ -0,0 +1,9 @@
+#[cfg(not(feature = "lock_timeout"))]
+mod lock_alias;
+#[cfg(not(feature = "lock_timeout"))]
+pub use lock_alias::*;
```

Comment:
You don't need `lock_alias.rs`. This entire file could be just:

```rust
#[cfg(feature = "lock_timeout")]
mod lock_timeout;

#[cfg(feature = "lock_timeout")]
pub use lock_timeout::RwLock;
#[cfg(feature = "lock_timeout")]
pub use lock_timeout::Mutex;


#[cfg(not(feature = "lock_timeout"))]
pub type RwLock<T> = tokio::sync::RwLock<T>;
#[cfg(not(feature = "lock_timeout"))]
pub type Mutex<T> = tokio::sync::Mutex<T>;
```

Reply:
Also please make sure that generic constraints of `tokio::sync::RwLock<T: ?Sized>` are honoured.